apis/v1/resourceSnapshot_types.go

```diff
@@ -0,0 +1,101 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v1
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope="Cluster",shortName=rss,categories={fleet-workload}
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:JSONPath=`.metadata.generation`,name="Gen",type=string
+// +kubebuilder:printcolumn:JSONPath=`.metadata.creationTimestamp`,name="Age",type=date
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ResourceSnapShot is used to store a snapshot of selected resources by a resource placement policy.
+// It is immutable. We may need to produce more than one resourceSnapshot for one ResourcePlacement to
+// get around the 1MB size limit of k8s objects.
+// Each snapshot must have a label "fleet.azure.com/snapshotGroup" with value as the snapshot index.
+// Each must have an annotation "fleet.azure.com/resource-hash" with value as the sha-256 hash
+// value of all the snapshots belong to the same snapshot index.
+type ResourceSnapShot struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// The desired state of ResourceSnapShot.
+	// +required
+	Spec ResourceSnapShotSpec `json:"spec"`
+
+	// The observed status of ResourceSnapShot.
+	// +optional
+	Status ResourceSnapShotStatus `json:"status,omitempty"`
+}
+
+// ResourceSnapShotSpec	defines the desired state of ResourceSnapShot.
+type ResourceSnapShotSpec struct {
+	// Index is the parent index of this resource snapshot. Each index can have multiple snapshots.
+	// All the snapshots with the same index have the same label "fleet.azure.com/snapshotGroup" that
+	// points to the index.
+	// +required
+	Index int32 `json:"index"`
```

Comment:
what's the index for the parent? is it a random number?
Can we use the parent name instead?

Why do we need the label in addition to this filed?

Reply:
We need the label so that we can select all the snapshot belongs to the same parent.  I think I have explained the pros vs cons about index vs name in the binding definition.